src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1075,6 +1080,45 @@ internal OneOrMany<SyntaxTree> GetSyntaxTreesByMappedPath(string mappedPath)
             }
         }
 
+        private class ContentHashComparer : IEqualityComparer<ReadOnlyMemory<byte>>
+        {
+            public bool Equals(ReadOnlyMemory<byte> x, ReadOnlyMemory<byte> y)
+            {
+                return x.Span.SequenceEqual(y.Span);
+            }
+
+            public int GetHashCode(ReadOnlyMemory<byte> obj)
+            {
+                return BinaryPrimitives.ReadInt32LittleEndian(obj.Span);
+            }
+
+            public static ContentHashComparer Instance { get; } = new ContentHashComparer();
+        }
+
+        internal OneOrMany<SyntaxTree> GetSyntaxTreesByContentHash(ReadOnlyMemory<byte> contentHash)
```

Comment:
```suggestion
        internal OneOrMany<SyntaxTree> GetSyntaxTreesWithContentHash(ReadOnlyMemory<byte> contentHash)
```

Reply:
Sigh ... this is a place where we should be taking `ReadOnlySpan<byte>` but can't because of the dictionary usage. One day though we will be able to do this. 

https://github.com/dotnet/runtime/issues/27229

Reply:
Not planning to make the method name change at this time.